### PR TITLE
Module

### PR DESCRIPTION
**Pull request by david-wirelab**

### CODE REVIEW

File: tests/vpcpeering_test.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import unittest
+from runner import Runner
+
+
+class TestE2E(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.snippet = """
+
+            provider "aws.source" {
```
Comment (chrisns) [inline tests/vpcpeering_test.py]: use the same `alias` syntax as we have in dq-tf-infra/main.tf

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  peer_owner_id = "${var.vpc_dest_account_id}"
+  peer_vpc_id   = "${var.vpc_dest_vpc_id}"
+  vpc_id        = "${var.vpc_source_vpc_id}"
+
+  tags {
+    Side = "Requester"
+  }
+
+  tags = {
+    Name = "${format("%s - %s", var.vpc_source_name, var.vpc_dest_name)}"
```
Comment (chrisns) [inline main.tf]: you should be able to get the name with `data`

```hcl
data "aws_vpc" "dest" {
  id = "${var. vpc_dest_vpc_id}"
}

data "aws_vpc" "source" {
  id = "${var. vpc_source_vpc_id}"
}

// then:
  Name = "${format("%s - %s", data. aws_vpc.source.name, data. aws_vpc.dest.name)}"
```

**Review by chrisns — CHANGES_REQUESTED** [review]

File: main.tf (hunk 2 above)
Comment (david-wirelab) [reply main.tf]: data provider does not seem to pull in region from root hence test is failing with error: "provider.aws: "region": required field is not set"

File: tests/vpcpeering_test.py (hunk 1 above)
Comment (david-wirelab) [reply tests/vpcpeering_test.py]: changed

File: main.tf (hunk 2 above)
Comment (chrisns) [reply main.tf]: Just pushed a fix for that

Comment (chrisns) [reply main.tf]: the issue was you not defining a provider block in the module which was the same gotcha we had before when everything deployed into the CI account

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+data "aws_vpc" "source" {
+  id = "${var.vpc_source_vpc_id}"
+}
+
+resource "aws_vpc_peering_connection" "request" {
+  provider = "aws.source"
+
+  auto_accept   = "false"
+  peer_owner_id = "${var.vpc_dest_account_id}"
```
Comment (chrisns) [inline main.tf]: account ID should come from a `data` see previous comment

**Review by chrisns — CHANGES_REQUESTED** [review]

File: tests/vpcpeering_test.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+             }
+
+            module "vpcpeering" {
+              source = "./mymodule"
+
+              providers = {
+                aws.source = "aws.APPS"
+                aws.dest = "aws.MOCK"
+              }
+            vpc_dest_account_id = "12345"
```
Comment (chrisns) [inline tests/vpcpeering_test.py]: you can remove this now?

File: variable.tf [hunk 5]
```diff
@@ -0,0 +1,5 @@
+variable "vpc_dest_account_id" {}
```
Comment (chrisns) [inline variable.tf]: and this can go also?

**Review by chrisns — CHANGES_REQUESTED** [review]

**Review by chrisns — DISMISSED** [review]

**Review by chrisns — DISMISSED** [review]

**Review by chrisns — APPROVED** [review]

**Pull request merged**